internal/util/util.go

```diff
@@ -115,15 +117,32 @@ func CheckDesiredBundleTemplate(existingBundle *rukpakv1alpha1.Bundle, desiredBu
 	bundle := existingBundle.DeepCopy()
 	delete(bundle.Labels, CoreOwnerKindKey)
 	delete(bundle.Labels, CoreOwnerNameKey)
-	return equality.Semantic.DeepEqual(bundle.Spec, desiredBundle.Spec) &&
-		equality.Semantic.DeepEqual(existingBundle.Annotations, desiredBundle.Annotations) &&
-		equality.Semantic.DeepEqual(bundle.Labels, desiredBundle.Labels)
+
+	if !equality.Semantic.DeepEqual(existingBundle.Annotations, desiredBundle.Annotations) ||
+		!equality.Semantic.DeepEqual(bundle.Labels, desiredBundle.Labels) {
+		return false
+	}
+
+	// Split up the bundle name into the BI name + hash.
+	// Since bundles are immutable, the name of the bundle is guaranteed to conform to what it was upon creation.
+	hash := strings.Split(bundle.GetName(), "-")[1]
+
+	hasher := fnv.New32a()
+	DeepHashObject(hasher, desiredBundle.Spec)
+	result := rand.SafeEncodeString(fmt.Sprint(hasher.Sum32()))[:10]
```

Comment:
not really happy with the arbitrary indexing here, but otherwise do we run into issues where the size of the hash overflows the limit for the size of a kube object (or even our webhook at 40 chars)?